src/Properties/Country.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace GingerPluginSdk\Properties;
+
+use GingerPluginSdk\Bases\BaseField;
+use GingerPluginSdk\Helpers\FieldsValidatorTrait;
+use GingerPluginSdk\Interfaces\ValidateFieldsInterface;
+
+final class Country extends BaseField implements ValidateFieldsInterface
+{
+    use FieldsValidatorTrait;
+
+    public function __construct($value)
+    {
+        parent::__construct("country");
+        $this->set($value);
+    }
+
+    /**
+     * @throws \GingerPluginSdk\Exceptions\OutOfPatternException
+     */
+    public function validate($value)
+    {
+        $this->validatePattern(
+            value: $value,
+            pattern: "/^[a-zA-Z]{2}$/",
+            valid: ['UA,NL,BE,FR']
```

Comment:
Personally I think that you should not check the countries here. There are way more valid countries then provided here and the additions are made externally. You could check the format (two uppercase letters).
I think the valid is also passed incorrectly 'UA,NL,BE,FR' instead of 'UA', 'NL', 'BE', 'FR'

Reply:
> Personally I think that you should not check the countries here. There are way more valid countries then provided here and the additions are made externally. You could check the format (two uppercase letters).

That's what I'm doing, I'm just using pattern check. The valid property is an example, as a hint, which will be printed on the screen in the Exception message.

> I think the valid is also passed incorrectly 'UA,NL,BE,FR' instead of 'UA', 'NL', 'BE', 'FR'

Looks like more correctly than my variant.